src/main/java/com/azure/cosmos/examples/analyticalcontainercrud/async/AnalyticalContainerCRUDQuickstartAsync.java

```diff
@@ -0,0 +1,214 @@
+
+    private CosmosAsyncClient client;
+
+    private final String databaseName = "AzureSampleFamilyDB";
+    private final String containerName = "FamilyContainer";
+
+    private CosmosAsyncDatabase database;
+    private CosmosAsyncContainer container;
+
+    protected static Logger logger = LoggerFactory.getLogger(AnalyticalContainerCRUDQuickstartAsync.class);
+
+    public void close() {
+        client.close();
+    }
+
+    /**
+     * Sample to demonstrate the following ANALYTICAL STORE container CRUD
+     * operations:
+     * -Create
+     * -Update throughput
+     * -Read by ID
+     * -Read all
+     * -Delete
+     */
+    public static void main(String[] args) {
+        AnalyticalContainerCRUDQuickstartAsync p = new AnalyticalContainerCRUDQuickstartAsync();
+
+        try {
+            logger.info("Starting ASYNC main");
+            p.containerCRUDDemo();
+            logger.info("Demo complete, please hold while resources are released");
+        } catch (Exception e) {
+            e.printStackTrace();
+            logger.error("Cosmos getStarted failed", e);
+        } finally {
+            logger.info("Closing the client");
+            p.shutdown();
+        }
+    }
+
+    private void containerCRUDDemo() throws Exception {
+
+        logger.info("Using Azure Cosmos DB endpoint: {}", AccountSettings.HOST);
+
+        // Create async client
+        client = new CosmosClientBuilder()
+                .endpoint(AccountSettings.HOST)
+                .key(AccountSettings.MASTER_KEY)
+                .consistencyLevel(ConsistencyLevel.EVENTUAL)
+                .contentResponseOnWriteEnabled(true)
+                .buildAsyncClient();
+
+        createDatabaseIfNotExists();
+        createContainerIfNotExists();
+        updateContainerThroughput();
+
+        readContainerById();
+        readAllContainers();
+
+        // deleteAContainer() is called at shutdown()
+
+    }
+
+    // Database Create
+    private void createDatabaseIfNotExists() throws Exception {
+        logger.info("Create database {} if not exists...", databaseName);
+
+        // Create database if not exists - this is async but we block to make sure
+        // database and containers are created before sample runs the CRUD operations on
+        // them
+        CosmosDatabaseResponse databaseResponse = client.createDatabaseIfNotExists(databaseName).block();
+        database = client.getDatabase(databaseResponse.getProperties().getId());
+
+        logger.info("Done.");
+    }
+
+    // Container create
+    private void createContainerIfNotExists() throws Exception {
+        logger.info("Create container " + containerName + " if not exists.");
+
+        // Create container if not exists
+        CosmosContainerProperties containerProperties = new CosmosContainerProperties(containerName, "/lastName");
+
+        // Set analytical store properties
+        containerProperties.setAnalyticalStoreTimeToLiveInSeconds(-1);
```

Comment:
this will require the analytical store to be enabled on the container, else it might throw exception I think. 


Does this example purposely only targeting for container with analytical store enabled?

Reply:
This action results in analytical store being enabled. It is copied from sync sample that already existed :-)

Reply:
ah right, this is the path of creating container..